common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -883,6 +918,25 @@ def answer_available(self):
 
         return False
 
+    def correctness_available(self):
+        """
+        Is the user allowed to see whether she's answered correctly?
+
+        Limits access to the correct/incorrect flags, messages, and problem score.
+        """
+        if self.show_correctness == SHOW_CORRECTNESS.NEVER:
+            return False
+        elif self.runtime.user_is_staff:
+            # This is after the 'never' check because admins can see correctness
```

Comment:
@cahrens @srpearce

> Thinking about `SHOWANSWER`, the default value is `CLOSED` (show the answer only after the learner has used up all his attempts to answer the problem or if the due date has passed), so course staff will also need to set `SHOWANSWER` to `NEVER` or `PAST_DUE` on every individual problem. That's unfortunate-- I was hoping that this subsection setting would remove the need to set something on every problem. Although perhaps these courses will set a coursewide default for `SHOWANSWER` in Advanced Settings? 

That's a very good point.  I can see three options here:

1. Leave the code as-is, and document the course-wide `SHOWANSWER` setting alongside the new `show_correctness` setting, with a recommendation to use `NEVER` or `PAST_DUE` as the course default.
1. Add `SHOWANSWER` to the subsection Visibility tab as well, so course authors can set it to `NEVER` or `PAST_DUE` for sections where `show_correctness` is also being withheld.
1. Make `show_correctness` also overwrite `SHOWANSWER` when `not correctness_available()`. 

Of these, I like the third option best. We're already suppressing adaptive hints, and I think the umbrella of "hide correctness/results" can logically extend to the "answer" visibility as well.

What do you think?  Is this too big a change to get approved by our merge deadline (21 Apr)?